src/data/moves/move.ts

```diff
@@ -339,6 +339,26 @@ export default class Move implements Localizable {
           return true;
         }
         break;
+      case PokemonType.FIRE:
+        if (move.id === Moves.WILL_O_WISP) {
+          return true;
+        }
+        break;
+      case PokemonType.ELECTRIC:
+        if (move.id === Moves.THUNDER_WAVE) {
+          return true;
+        }
+        break;
+      case PokemonType.POISON:
+        if (move.id === Moves.TOXIC) {
+          return true;
+        }
+        break;
+      case PokemonType.STEEL:
+        if (move.id === Moves.POISON_GAS) {
+          return true;
+        }
+        break;
```

Comment:
```suggestion
      case PokemonType.POISON:
      case PokemonType.STEEL:
        if ([Moves.TOXIC, Moves.POISON_GAS, Moves.POISON_POWDER].includes(move.id)) {
          return true;
        }
        break;
```
I'm not sure this is the best method though, ideally it would tie in to `Pokemon.canSetStatus()` or something.

Reply:
Thank you for reviewing my code. I'll change my code to implement what you suggested.